src/vs/base/browser/ui/iconLabel/iconLabel.ts

```diff
@@ -155,18 +162,74 @@ export class IconLabel extends Disposable {
 
 			if (this.descriptionNode instanceof HighlightedLabel) {
 				this.descriptionNode.set(description || '', options ? options.descriptionMatches : undefined);
-				if (options?.descriptionTitle) {
-					this.descriptionNode.element.title = options.descriptionTitle;
-				} else {
-					this.descriptionNode.element.removeAttribute('title');
-				}
+				this.setupHover(this.descriptionNode.element, options?.descriptionTitle);
 			} else {
 				this.descriptionNode.textContent = description || '';
-				this.descriptionNode.title = options?.descriptionTitle || '';
+				this.setupHover(this.descriptionNode.element, options?.descriptionTitle || '');
 				this.descriptionNode.empty = !description;
 			}
 		}
 	}
+
+	private setupHover(htmlElement: HTMLElement, tooltip: string | Promise<IMarkdownString | string | undefined> | undefined): void {
```

Comment:
Since this method is long, should we split it up into `setupCustomHover` and `setupNativeHover`? This makes it easier to follow on what is happening.